src/libs/actions/IOU.js

```diff
@@ -122,82 +134,277 @@ function createIOUTransaction(params) {
 }
 
 /**
- * Creates IOUSplit Transaction
+ * Build the Onyx data and IOU split necessary for splitting a bill with 3+ users.
+ * 1. Build the optimistic Onyx data for the group chat, i.e. chatReport and iouReportAction creating the former if it doesn't yet exist.
+ * 2. Loop over the group chat participant list, building optimistic or updating existing chatReports, iouReports and iouReportActions between the user and each participant.
+ * We build both Onyx data and the IOU split that is sent as a request param and is used by Auth to create the chatReports, iouReports and iouReportActions in the database.
+ * The IOU split has the following shape:
+ *  [
+ *      {email: 'currentUser', amount: 100},
+ *      {email: 'user2', amount: 100, iouReportID: '123', chatReportID: '123', transactionID: '123', reportActionID: '123', clientID: '123'},
+ *      {email: 'user3', amount: 100, iouReportID: '123', chatReportID: '123', transactionID: '123', reportActionID: '123', clientID: '123'}
```

Comment:
NAB: This may be a very noobish thing, but `iouReportID`, `chatReportID`, `transactionID`, `reportActionID` and `clientID` are all different between different participants, right? I think it would be better if this example had different values unless they are really the same :P

Reply:
good shout